contributing: add guide of signing CLA (#4160)
pingcap/docs: .github/pull_request_template.md

<!--Thanks for your contribution to TiDB documentation. Please answer the following questions.-->

> ### First-time contributors' checklist

Reviewer: Suggest making this item the first priority to check. PTAL @lilin90

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

<!--Tell us what you did and why.-->

### Which TiDB version(s) do your changes apply to? (Required)

<!-- Fill in "x" in [] to tick the checkbox below.-->

- [ ] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

<!--Reference link(s) will help reviewers review your PR quickly.-->

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases

> - [ ] Need modification after applied to another branch <!-- If yes, please add the label "version-specific-changes-required"-->

Reviewer: ```suggestion
- [ ] Need modification after applied to another branch <!-- If yes, please leave a comment "/label version-specific-changes-required"-->
```
